### PR TITLE
feat(ST-106): unenroll student from classroom by authorized teacher

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Added endpoint for unenrolling student from classroom. 
- Implemented the logic in the service layer by adding method named removeStudent
- Updated the enrollment module to use the user repository
## Associated Ticket Link(s)

- https://trello.com/c/pv9Dm6VA

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/enrollments/enrollments.service.ts [hunk 1]
```diff
@@ -46,4 +48,21 @@ export class EnrollmentsService {
...
+
+  async removeStudent(userId: number, deleteEnrollDto: CreateEnrollmentDto): Promise<void> {
+    const { studentId, classroomId } = deleteEnrollDto;
+
+    const user = await this.userRepository.findOneOrFail({ id: userId });
+    const classroom = await this.classroomRepository.findOneOrFail({ id: classroomId });
+
+    if (user.teacher && user.teacher.id !== classroom.teacher.id) {
+      throw new UnauthorizedException("You are not authorized to remove this student!");
+    }
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.service.ts:60]: I think abstracting classroom teacher and student validation to a separate method makes sense here. Please make a tech debt ticket to refactor this validation into a new method and replace wherever this check is necessary with that method.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

File: src/enrollments/enrollments.controller.ts [hunk 2]
```diff
@@ -18,4 +19,13 @@ export class EnrollmentsController {
...
   }
+
+  @Roles(EUserType.TEACHER)
+  @Delete()
+  unenrollStudent(
+    @CurrentUser() user: { id: number },
+    @Body() deleteEnrollDto: CreateEnrollmentDto,
+  ): Promise<void> {
+    return this.enrollmentsService.removeStudent(user.id, deleteEnrollDto);
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/enrollments/enrollments.controller.ts:30]: I think this should be inside the classrooms controller

Comment (ghost) [reply src/enrollments/enrollments.controller.ts:30]: Bhaiya, all the enrollments and unenrollments of students are kept in this controller. I have done this for ease of use.

Comment (ishmam-iqbal-sazim) [reply src/enrollments/enrollments.controller.ts:30]: Moving the endpoints to classrooms controller wouldn't be much of a difference. You'd just have to call the same service method in that file.
And please follow REST principles and use endpoints such as: `classrooms/:classroomId/enroll`, 
`classrooms/:classroomId/students` etc.

Comment (ghost) [reply src/enrollments/enrollments.controller.ts:30]: Okay bhaiya. I will create a separate ticket for moving all the enrollment controller to classroom controller.

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**